Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -2043,6 +2060,10 @@ void qMRMLSegmentEditorWidget::setCurrentSegmentID(const QString segmentID)
 void qMRMLSegmentEditorWidget::onMasterVolumeNodeChanged(vtkMRMLNode* node)
 {
   Q_D(qMRMLSegmentEditorWidget);
+  if (!this->mrmlScene() || this->mrmlScene()->IsBatchProcessing())
```

Comment:
The master volume node selector has a "None" option, so if a node is deleted then that should not lead to any performance issue (the combobox will simply switch to the None option). Are you sure you need to ignore the event?

Reply:
I’m pretty sure it is just like the Slice views volume node combo box in that if the current selection is removed, it selects the last available index. It won’t select the “None” index. So when deleting the current selection of the master volume node combo box over and over it will keep triggering the event.

Reply:
@lassoan I have confirmed that deleting the current selection of the master volume node will choose the last available index. This selects another volume node as the master volume node when there are multiple volumes loaded in the scene. It only selects the "Select master volume to enable editing" option if there are no more volume nodes to choose from.

https://user-images.githubusercontent.com/15837524/170770942-e8dbf9eb-462e-4cb9-a079-2b9416a6cc99.mp4

Reply:
I've just tested this and I've found that the result of master volume deletion is different in batch-mode and non-batch-mode.

How to reproduce:

- Load [this scene](https://1drv.ms/u/s!Arm_AFxB9yqHyJd5y3S3XT5BShwTvw?e=lZRaZk) (contains 3 segmentation nodes, each with a different master volume)
- Go to Segment editor module
- Run this script:

```
scene = slicer.mrmlScene
scene.StartState(scene.BatchProcessState)
slicer.mrmlScene.RemoveNode(getNode('MRBrainTumor1'))
scene.EndState(scene.BatchProcessState)
```

Result: master volume is now `Select master volume to enable editing`

- Restart Slicer, load the same scene
- Run this script:
- ```
scene = slicer.mrmlScene
slicer.mrmlScene.RemoveNode(getNode('MRBrainTumor1'))
```

Result: master volume is now `CTChest`

The problem may be that the `setMasterVolumeNode` has been completely missed. I don't think we can skip it, we can just delay it until the batch processing is completed.

It probably happens because even though we return early from `onMasterVolumeNodeChanged`, there is still a call to `updateWidgetFromMRML` (due to node references are being changed, which invokes a parameter node modified event).

Both selecting another random volume and setting master volume to nullptr makes sense (I think I prefer the latter), but what is not OK to have different behavior in batch and non-batch mode.